torch/distributed/_composable/fsdp/_fsdp_param_group.py

```diff
@@ -374,6 +378,7 @@ def post_backward(self, *unused: Any):
                 self.comm_ctx.reduce_scatter_state = None
             (
                 reduce_scatter_input,
+                self._reduce_output,
```

Comment:
this breaks optimizer in backward :( ugh, big pain

Reply:
and CPU offloading :(

Reply:
The issue is that there is not a good/simple point at which to have the default stream wait for the `post_reduce_event` in the middle of backward :/